modules/renter/contractor/editor.go

```diff
@@ -116,6 +116,7 @@ func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
 	he.contractor.persist.update(updateUploadRevision{
 		NewRevisionTxn:     contract.LastRevisionTxn,
 		NewSectorRoot:      sectorRoot,
+		NewSectorIndex:     len(contract.MerkleRoots) - 1,
```

Comment:
it's a little confusing, but what happens is that `contract.MerkleRoots` gets appended to before this call. So the length is already 1+ the old length.